go/vt/vttablet/tabletserver/vstreamer/vstreamer.go

```diff
@@ -1046,6 +1053,75 @@ func (vs *vstreamer) extractRowAndFilter(plan *streamerPlan, data []byte, dataCo
 		}
 		pos += l
 
+		// If the column is a CHAR based type with a binary collation (e.g. utf8mb4_bin) then the
+		// actual column type is included in the second byte of the event metadata while the
+		// event's type for the field is BINARY. This is true for ENUM and SET types.
+		var mysqlType uint16
+		if sqltypes.IsQuoted(plan.Table.Fields[colNum].Type) {
+			mysqlType = plan.TableMap.Metadata[colNum] >> 8
+		}
+		// Convert the integer values in the binlog event for any SET and ENUM fields into their
+		// string representations.
+		if plan.Table.Fields[colNum].Type == querypb.Type_ENUM || mysqlType == mysqlbinlog.TypeEnum {
```

Comment:
probably a good ida to extract out the enum and set specific code into separate methods.

Reply:
Done.